dungeon/src/task/quizquestion/QuizDialogDesign.java

```diff
@@ -32,15 +30,14 @@ public static VerticalGroup createAnswerButtons(Skin skin, QuizQuestion quizQues
                     case SINGLE_CHOICE -> skin.get("radio", CheckBox.CheckBoxStyle.class);
                     default -> skin.get("default", CheckBox.CheckBoxStyle.class);
                 };
-        Arrays.stream(quizQuestion.answers())
-                .filter(
-                        answer ->
-                                answer.type() != QuizQuestionContent.QuizQuestionContentType.IMAGE)
+        quizQuestion
+                .contentStream()
+                .map(answer -> (QuizQuestion.QuizContent) answer)
+                .filter(answer -> answer.type() != QuizQuestion.QuizContentType.IMAGE)
```

Comment:
wenn ich das sehe: sollte der type (und auch das enum) evtl. besser in der basisklasse sein? das spart dann das wilde gecaste ...

Reply:
Welche Klasse meinst du mit Basisklasse?

Reply:
quizquestion leitet von task ab ...

Reply:
Achse, irgendwie dachte ich du meinst die Basisklasse des enumeriert und war etwas verwirrt.

Reply:
Solange wie Vererbung nutzen, finde ich das so besser. So kann ich spezifisch pro aufgabentypen definieren welche Typen vom Contest es dafür gibt und vermeide "ungültige" Aufgabentypen Contenttypen Kombinationen

Reply:
jaaa, schon. aber es fühlt sich nicht richtig an, und das komplexe uml plus bereits hier im beispiel der zwang zum extensiven casten unterstützen das bauchgefühl. die zuordnung kannst du ja auch über methoden konsistent machen.

Reply:
Ich mach aus der Diskussion auch mal ein neues Issue, dann können wir hier erstmal weitermachen und der Gedanke geht nicht verloren. 
Mein Bauchgefühl sagt mir, um irgendeine Art von Typchchecking komm ich eh nicht drum herum. 

Hier lässt sich vermutlich besser eine Entscheidung treffen, wenn die anderen Aufgabentypen auch umgesetzt sind.

Reply:
#863 
Der Titel ist irgendwie nicht so richtig 100% griffig, aber mir fehlt grade die kreativität. 
ich hab auch nochmal dein Review Kommentar verlinkt. Hängt für mich zusammen.